### DISCUSSION TITLE
PDF Annotation Support

### DISCUSSION THREAD

**Discussion by tobias0409** [personalizedrefrigerator/js-draw#64] [Ideas & Feature Requests]
Hey, great plugin! I love it.

My idea would be to add PDF Annotation Support. Additionally I would love to have a way to select between whiteboard and e.g A4 paper size mode. 

If I choose the paper mode, then it would be nice to add additional pages and scroll through the pages. 

Moreover the ultimate dealbreaker would be to combine the PDF Annotation function with the possibilty to add own empty A4 sheets for adding notes when there is no space on the PDF and export it as PDF again.

**Comment by personalizedrefrigerator** [comment]
> My idea would be to add PDF Annotation Support.

I would also find this very useful.

An idea that was posted on the Joplin forum was to start by allowing the "insert image" tool to insert PDFs as images. To make this easier to support, I've opened a pull request on the Joplin repository: https://github.com/laurent22/joplin/pull/10177

**Reply by personalizedrefrigerator** [reply]
I was originally looking into PDF.js for this. At present, however, it doesn't seem to have a good API for adding/removing annotations.

I'm now looking into [MuPDF](https://github.com/ArtifexSoftware/mupdf.js). It's licensed under the AGPLv3, but I think that should be fine for use with Joplin.

**Reply by personalizedrefrigerator** [reply]
Related pull request: https://github.com/personalizedrefrigerator/js-draw/pull/67

**Comment by personalizedrefrigerator** [comment]
Related release: [v2.10.2](https://github.com/personalizedrefrigerator/joplin-plugin-freehand-drawing/releases/tag/v2.10.2).
